src/dataviews/dataview-model-base.js

```diff
@@ -161,15 +160,14 @@ module.exports = Model.extend({
 
   _setupAnalysisStatusEvents: function () {
     this._removeExistingAnalysisBindings();
-    this._analysis = this.getSource();
-    if (this._analysis) {
-      this._analysis.on('change:status', this._onAnalysisStatusChange, this);
+    if (this.getSource()) {
+      this.getSource().on('change:status', this._onAnalysisStatusChange, this);
     }
   },
 
   _removeExistingAnalysisBindings: function () {
-    if (!this._analysis) return;
-    this._analysis.off('change:status', this._onAnalysisStatusChange, this);
+    if (!this.get('source')) return;
```

Comment:
I'd try to use `getSource` instead of accessing the `source` attribute as much as possible.